### PR TITLE
Add tests for Extra.it

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#2192 

# Overview

- [x] I added a test suite for `cardano-wallet-test-utils` which could be useful
- [x] I added some basic tests for #2181

# Comments

### CODE REVIEW

File: lib/test-utils/test/Test/Hspec/ExtraSpec.hs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            $ beforeAll (return ())
+            $ anyIt "<test spec>" (const prop)
+
+        stripTime :: String -> String
+        stripTime = unlines
+                . filter (not . ("Finished in" `isPrefixOf`))
+                . filter (not . ("Randomized" `isPrefixOf`))
+                . lines
+    dynamically
+        :: IORef [Expectation]
```
Comment (Anviking) [inline lib/test-utils/test/Test/Hspec/ExtraSpec.hs]: should probably be
```suggestion
        :: IORef [IO ()]
```

**Review by rvl — APPROVED** [review]
Testing the tests? Crazy! :+1:

**Pull request merged**